cmake/compiler/host-gcc/target.cmake

```diff
@@ -9,10 +9,19 @@ find_program(CMAKE_RANLILB      ranlib )
 find_program(CMAKE_READELF      readelf)
 find_program(CMAKE_GDB          gdb    )
 
-set(CMAKE_ASM_FLAGS           -m32 )
-set(CMAKE_C_FLAGS             -m32 )
-set(CMAKE_CXX_FLAGS           -m32 )
-set(CMAKE_SHARED_LINKER_FLAGS -m32 ) # unused?
+# -march={pentium,lakemont,...} do not automagically imply -m32, so
+# adding it here.
+
+# There's only one 64bits ARCH (actually: -mx32). Let's exclude it to
```

Comment:
By x32 you mean the x32 ABI which has more registers than i386, right?
https://en.wikipedia.org/wiki/X32_ABI

But native_posix requires 32-bit always (so should be i386 compatible (and x32 I suppose)) and then you want to _not_ set -m32 if we are on x86-64? What am I missing?

Reply:
Yes native_posix is neither 64 bits nor x32, which is why it's unrelated to this fix.

> then you want to not set -m32 if we are on x86-64?

When compiling x86_64 with ZEPHYR_TOOLCHAIN_VARIANT=host right now, -m32 is (silently) overridden by -mx32 because -mx32 (luckily?) comes last. That doesn't seem right. This first part of the patch doesn't fix anything but it removes this element of luck.